lib/auth/grpcserver.go

```diff
@@ -2224,17 +2225,41 @@ func downgradeRoleToV6(r *types.RoleV6) (*types.RoleV6, bool, error) {
 			// If the role specifies any kubernetes resources, the V6 role will
 			// be unable to be used for kubernetes access because the labels
 			// will be empty and won't match anything.
-			downgraded.SetLabelMatchers(
-				types.Allow,
-				types.KindKubernetesCluster,
-				types.LabelMatchers{
-					Labels: types.Labels{},
-				},
-			)
-			// Clear out the allow list so that the V6 role doesn't include unknown
-			// resources in the allow list.
-			downgraded.SetKubeResources(types.Allow, nil)
-			restricted = true
+			hasRestritiveRules := false
+			for _, resource := range r.GetKubeResources(types.Allow) {
+				if resource.Kind != types.Wildcard ||
+					resource.Namespace != types.Wildcard ||
+					resource.Name != types.Wildcard ||
+					!slices.Equal(resource.Verbs, []string{types.Wildcard}) {
```

Comment:
I'd make it more explicit here
```suggestion
					(len(resource.Verbs) != 1 || resource.Verbs[0] != types.Wildcard) {
```

Reply:
(what if there's two verbs and they're both `"*"` tho?)

Reply:
You can't. `CheckAndSetDefaults` prevents you from adding `*` and any other verb